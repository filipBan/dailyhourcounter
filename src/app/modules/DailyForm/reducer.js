import {
  START_SAVING_DAY_DATA,
  FINISHED_SAVING_DAY_DATA,
  ERROR_SAVING_DAY_DATA,
  UPDATE_BREAKS,
  UPDATE_HOURS,
  SET_TODAY_DATE,
  CLEAR_ALL_TIMES,
  REPLACE_DAY_DATA,
  RESET_DAY_DATA,
  calculateTimeWorked,
  calculateTotalBreaks
} from "./actions";

const initialState = {
  today: new Date(),
  hours: [
    {
      start: null,
      end: null
    }
  ],
  breaks: [
    {
      start: null,
      end: null
    }
  ],
  workedMinutes: 0,
  breakMinutes: 0,
  savingData: false,
  error: null
};

const today = (state = initialState, action) => {
  switch (action.type) {
    case SET_TODAY_DATE:
      return {
        ...state,
        today: action.date
      };
    case UPDATE_HOURS:
      return {
        ...state,
        hours: [
          {
            start:
              action.timeType === "workStart"
                ? action.amount
                : state.hours[0].start,
            end:
              action.timeType === "workEnd" ? action.amount : state.hours[0].end
          }
        ],
        workedMinutes:
          action.timeType === "workEnd"
            ? calculateTimeWorked(state.hours[0].start, action.amount)
            : action.timeType === "workStart" && state.workEnd
            ? calculateTimeWorked(action.amount, state.hours[0].end)
            : state.workedMinutes
      };
    case UPDATE_BREAKS:
      return {
        ...state,
        breaks: [
          {
            start:
              action.timeType === "breakStart"
                ? action.amount
                : state.breaks[0].start,
            end:
              action.timeType === "breakEnd"
                ? action.amount
                : state.breaks[0].end
          }
        ],
        breakMinutes:
          action.timeType === "breakEnd"
            ? calculateTotalBreaks(state.breaks[0].start, action.amount)
            : action.timeType === "breakStart" && state.breakEnd
            ? calculateTotalBreaks(action.amount, state.breaks[0].end)
            : state.breakMinutes
      };
    case START_SAVING_DAY_DATA:
      return {
        ...state,
        savingData: true
      };
    case FINISHED_SAVING_DAY_DATA:
      return {
        ...state,
        savingData: false
      };
    case ERROR_SAVING_DAY_DATA:
      return {
        ...state,
        savingData: false,
        error: action.error
      };
    case CLEAR_ALL_TIMES:
      return {
        ...state,
        hours: [
          {
            start: null,
            end: null
          }
        ],
        breaks: [
          {
            start: null,
            end: null
          }
        ],
        workedMinutes: 0,
        breakMinutes: 0
      };

    case REPLACE_DAY_DATA:
      return {
        ...state,
        ...action.payload
      };

    case RESET_DAY_DATA:
      return initialState;
    default:
      return state;
  }
};

export default today;
