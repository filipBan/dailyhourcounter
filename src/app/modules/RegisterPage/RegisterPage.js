import React, { Component } from "react";
import { Redirect, Link } from "react-router-dom";

import Input from "@material-ui/core/Input";
import Card from "@material-ui/core/Card";

import styled from "styled-components";

import Button from "../../components/Button";

const Container = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  background-color: #89cff0;
  height: auto;
  min-height: 100vh;
  width: 100%;
`;

const Logo = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  flex: 1;
  font-family: "Bungee", cursive;
  font-size: 4rem;
`;

const FormSection = styled.div`
  flex: 3;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: flex-start;
  @media only screen and (min-width: 800px) {
    & {
      flex: 4;
    }
  }
`;

const TermsLinks = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  font-size: 1.4rem;
  text-align: center;
  margin-bottom: 2rem;

  a {
    margin-top: 1rem;
  }
`;

const FormContainer = styled(Card)`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  width: 30rem;
  padding: 2rem 3rem;
`;

const Form = styled.form`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
  margin-bottom: 2rem;
`;

const StyledInput = styled(Input)`
  input {
    font-size: 1.6rem;
    padding-top: 2.5rem;
  }
`;

class RegisterPage extends Component {
  state = {
    userName: "",
    wages: "",
    email: "",
    password: "",
    confirmPassword: ""
  };

  submitForm(e) {
    e.preventDefault();
    const { userName, wages, email, password, confirmPassword } = this.state;
    if (
      userName &&
      wages &&
      email &&
      password &&
      confirmPassword &&
      password === confirmPassword
    ) {
      this.props.registerNewAccount(this.state);
    }
  }

  handleInputChange(field, event) {
    const { value } = event.target;
    this.setState({ [field]: value });
  }

  render() {
    if (this.props.auth.isLoggedIn) {
      return <Redirect to="/today" />;
    }

    const { userName, wages, email, password, confirmPassword } = this.state;
    return (
      <Container>
        <Logo>
          <span>Daily</span>
          <span>Hours</span>
        </Logo>
        <FormSection>
          <FormContainer>
            <Form onSubmit={event => this.submitForm(event)}>
              <StyledInput
                type="string"
                value={userName}
                onChange={e => this.handleInputChange("userName", e)}
                placeholder="user name"
                fullWidth
                required
              />
              <StyledInput
                type="number"
                value={wages}
                onChange={e => this.handleInputChange("wages", e)}
                placeholder="current hourly salary"
                fullWidth
                required
              />
              <StyledInput
                type="email"
                value={email}
                onChange={e => this.handleInputChange("email", e)}
                placeholder="email"
                fullWidth
                required
              />
              <StyledInput
                type="password"
                value={password}
                onChange={e => this.handleInputChange("password", e)}
                placeholder="password"
                fullWidth
                required
              />
              <StyledInput
                type="password"
                value={confirmPassword}
                onChange={e => this.handleInputChange("confirmPassword", e)}
                placeholder="confirm password"
                fullWidth
                required
              />
              <Button
                type="submit"
                color="primary"
                variant="contained"
                fullWidth
                style={{ marginTop: "2rem" }}
              >
                Register
              </Button>
            </Form>
            <TermsLinks>
              <span>
                By registering you automatically accept the following:
              </span>
              <Link to="/register">
                <span>Terms and Conditions</span>
              </Link>
              <Link to="/register">
                <span>Privacy policy</span>
              </Link>
            </TermsLinks>
            <TermsLinks>
              Already have an account?
              <Link to="/">Login</Link>
            </TermsLinks>
          </FormContainer>
        </FormSection>
      </Container>
    );
  }
}

export default RegisterPage;
