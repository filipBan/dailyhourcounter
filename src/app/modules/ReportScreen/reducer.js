const initialState = {};

const reports = (state = initialState, action) => {
  switch (action.type) {
    default:
      return state;
  }
};

export default reports;
