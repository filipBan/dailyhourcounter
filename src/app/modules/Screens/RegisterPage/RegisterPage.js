import React, { Component } from "react";
import { Redirect, Link } from "react-router-dom";

import LinearProgress from "@material-ui/core/LinearProgress";

import Button from "../../../components/Button";
import Snackbar from "../../../components/Snackbar";
import AuthInput from "../../../components/AuthInput";

import {
  Container,
  Logo,
  FormSection,
  BottomLinks,
  FormContainer,
  Form,
  Progress
} from "../../../components/StyledComponents/LoginRegister";

class RegisterPage extends Component {
  submitForm(e) {
    e.preventDefault();
    const { userName, wages, email, password, confirmPassword } = this.props;
    if (
      userName &&
      wages &&
      email &&
      password &&
      confirmPassword &&
      password === confirmPassword
    ) {
      return this.props.registerNewAccount(this.props);
    } else if (password !== confirmPassword) {
      return this.props.authError("Passwords don't match");
    }
  }

  handleInputChange(field, event) {
    const { value } = event.target;
    this.props.updateInput(field, value);
  }

  render() {
    if (this.props.isLoggedIn) {
      return <Redirect to="/today" />;
    }

    const {
      userName,
      wages,
      email,
      password,
      confirmPassword,
      loading,
      error,
      clearAuthErrors
    } = this.props;

    return (
      <Container>
        <Logo fontSize="4rem">
          <span>Daily</span>
          <span>Hours</span>
        </Logo>
        <FormSection>
          <FormContainer>
            <Progress>{loading && <LinearProgress />}</Progress>
            <Form onSubmit={event => this.submitForm(event)}>
              <AuthInput
                type="userName"
                value={userName}
                onChange={(type, e) => this.handleInputChange(type, e)}
                paddingtop="2rem"
              />
              <AuthInput
                type="number"
                value={wages}
                onChange={(type, e) => this.handleInputChange("wages", e)}
                paddingtop="2rem"
              />
              <AuthInput
                type="email"
                value={email}
                onChange={(type, e) => this.handleInputChange(type, e)}
                paddingtop="2rem"
              />
              <AuthInput
                type="password"
                value={password}
                onChange={(type, e) => this.handleInputChange(type, e)}
                paddingtop="2rem"
              />
              <AuthInput
                type="password"
                value={confirmPassword}
                onChange={(type, e) =>
                  this.handleInputChange("confirmPassword", e)
                }
                paddingtop="2rem"
              />
              <Button
                type="submit"
                color="primary"
                variant="contained"
                fullWidth
                style={{ marginTop: "2rem" }}
                disabled={loading}
              >
                Register
              </Button>
            </Form>
            <BottomLinks marginBottom="2rem">
              <span>
                By registering you automatically accept the following:
              </span>
              <Link to="/register">
                <span>Terms and Conditions</span>
              </Link>
              <Link to="/register">
                <span>Privacy policy</span>
              </Link>
            </BottomLinks>
            <BottomLinks marginBottom="2rem">
              Already have an account?
              <Link to="/">Login</Link>
            </BottomLinks>
          </FormContainer>
        </FormSection>
        <Snackbar error={error} onClose={clearAuthErrors} />
      </Container>
    );
  }
}

export default RegisterPage;
