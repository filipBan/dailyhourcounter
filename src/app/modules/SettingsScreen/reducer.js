import { UPDATE_WAGES_INPUT } from "./actions";

const initialState = {
  wages: ""
};

const settings = (state = initialState, action) => {
  switch (action.type) {
    case UPDATE_WAGES_INPUT:
      return {
        ...state,
        wages: action.wages
      };

    default:
      return state;
  }
};

export default settings;
