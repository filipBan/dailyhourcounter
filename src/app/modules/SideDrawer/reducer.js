import { TOGGLE_DRAWER } from "./actions";

const initialState = {
  visible: false
};

const sideDrawer = (state = initialState, action) => {
  switch (action.type) {
    case TOGGLE_DRAWER:
      return {
        visible: !state.visible
      };
    default:
      return state;
  }
};
export default sideDrawer;
