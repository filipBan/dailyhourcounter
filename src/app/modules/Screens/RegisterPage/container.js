import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import RegisterPage from "./RegisterPage";

import {
  registerNewAccount,
  updateInput,
  authError
} from "../../../store/auth/actions";

const mapStateToProps = state => ({
  isLoggedIn: state.auth.isLoggedIn,
  loading: state.auth.loading,
  error: state.auth.error,
  email: state.auth.register.email,
  password: state.auth.register.password,
  confirmPassword: state.auth.register.confirmPassword,
  userName: state.auth.register.userName,
  wages: state.auth.register.wages
});

const mapDispatchToProps = dispatch => ({
  registerNewAccount: props => dispatch(registerNewAccount(props)),
  updateInput: (field, value) =>
    dispatch(updateInput("register", field, value)),
  authError: error => dispatch(authError(error))
});

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(RegisterPage)
);
