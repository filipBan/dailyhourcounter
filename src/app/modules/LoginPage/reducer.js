import {
  AUTH_SUCCESS,
  AUTH_FAIL,
  AUTH_LOGOUT,
  AUTH_START,
  AUTH_END,
  UPDATE_INPUT,
  UPDATE_USER_DATA
} from "./actions";
import { REPLACE_DAY_DATA } from "../DailyForm/actions";
import { UPDATE_WAGES } from "../SettingsScreen/actions";

const initialState = {
  isLoggedIn: false,
  name: "",
  uid: "",
  wages: 0,
  loading: false,
  error: null,
  email: "",
  password: "",
  emailVerified: false,
  checkingAuthState: true
};

const auth = (state = initialState, action) => {
  switch (action.type) {
    case AUTH_START:
      return { ...state, loading: true };
    case AUTH_END:
      return {
        ...initialState,
        checkingAuthState: false
      };
    case AUTH_SUCCESS:
      const user = {
        isLoggedIn: true,
        emailVerified: action.payload.emailVerified,
        email: action.payload.email,
        uid: action.payload.uid,
        loading: false,
        checkingAuthState: false
      };
      return { ...state, ...user };
    case AUTH_FAIL:
      return { ...state, loading: false, error: action.payload };
    case AUTH_LOGOUT:
      return initialState;
    case UPDATE_INPUT:
      return {
        ...state,
        [action.payload.field]: action.payload.value
      };
    case UPDATE_USER_DATA:
      return {
        ...state,
        name: action.user.userName,
        wages: action.user.wages
      };
    case REPLACE_DAY_DATA:
      return {
        ...state,
        wages: action.payload.wages || state.wages
      };
    case UPDATE_WAGES:
      return {
        ...state,
        wages: action.wages
      };
    default:
      return state;
  }
};

export default auth;
