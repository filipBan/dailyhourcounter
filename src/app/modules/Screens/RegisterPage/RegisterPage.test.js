import React, { Suspense } from "react";
import { applyMiddleware, createStore } from "redux";
import { Provider, connect } from "react-redux";
import { render, fireEvent, cleanup } from "react-testing-library";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import { MuiPickersUtilsProvider } from "material-ui-pickers";
import DateFnsUtils from "@date-io/date-fns";
import thunk from "redux-thunk";

import RegisterPage from "./RegisterPage";
import reducer from "../../../store/reducers";
import {
  clearAuthErrors,
  registerNewAccount,
  updateInput,
  authError
} from "../../../store/auth/actions";

const initialState = Object.freeze({
  today: {},
  auth: {
    isLoggedIn: false,
    name: "",
    uid: "",
    wages: 0,
    loading: false,
    error: null,
    emailVerified: false,
    checkingAuthState: true,
    login: {
      email: "",
      password: ""
    },
    register: {}
  },
  ui: {
    notification: {
      modalOpen: false,
      notification: {}
    },
    sideDrawer: {
      visible: false
    }
  },
  reports: {},
  settings: {}
});

const mapStateToProps = state => ({
  isLoggedIn: state.auth.isLoggedIn,
  loading: state.auth.loading,
  error: state.auth.error,
  email: state.auth.register.email,
  password: state.auth.register.password,
  confirmPassword: state.auth.register.confirmPassword,
  userName: state.auth.register.userName,
  wages: state.auth.register.wages
});

const mapDispatchToProps = dispatch => ({
  registerNewAccount: props => dispatch(registerNewAccount(props)),
  clearAuthErrors: () => dispatch(clearAuthErrors()),
  updateInput: (field, value) =>
    dispatch(updateInput("register", field, value)),
  authError: error => dispatch(authError(error))
});

const mergeProps = (stateProps, dispatchProps, ownProps) => ({
  ...stateProps,
  ...dispatchProps,
  ...ownProps
});

const ConnectedRegisterPage = connect(
  mapStateToProps,
  mapDispatchToProps,
  mergeProps
)(RegisterPage);

afterEach(cleanup);

function renderWithRedux(ui, store) {
  return {
    ...render(
      <MuiPickersUtilsProvider utils={DateFnsUtils}>
        <Provider store={store}>
          <Router history={createMemoryHistory({ initialEntries: ["/"] })}>
            <Suspense fallback={<div>Loading</div>}>{ui}</Suspense>
          </Router>
        </Provider>
      </MuiPickersUtilsProvider>
    ),
    store
  };
}

describe("Registration Page integration tests", () => {
  it("Can render with redux with defaults", () => {
    const store = createStore(reducer, initialState);
    renderWithRedux(<ConnectedRegisterPage />, store);
  });

  it("Should match the snapshot", () => {
    const store = createStore(reducer, initialState);
    const { container } = renderWithRedux(<ConnectedRegisterPage />, store);

    expect(container).toMatchSnapshot();
  });

  it("Should not allow to submit the form without filling out all required fields", () => {
    const store = createStore(reducer, initialState);
    const registerNewAccountMock = jest.fn();
    const { getByLabelText } = renderWithRedux(
      <ConnectedRegisterPage registerNewAccount={registerNewAccountMock} />,
      store
    );

    fireEvent.click(getByLabelText("register-submit-button"));

    expect(registerNewAccountMock).not.toHaveBeenCalled();
  });
});
