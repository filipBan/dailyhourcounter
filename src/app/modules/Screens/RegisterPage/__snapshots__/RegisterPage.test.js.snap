// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration Page integration tests Should match the snapshot 1`] = `
<div>
  <div
    class="sc-ifAKCX jqsrVw"
  >
    <header
      class="sc-EHOje fOdpYX"
      font-size="4rem"
    >
      <span>
        Daily
      </span>
      <span>
        Hours
      </span>
    </header>
    <main
      class="sc-bZQynM dwLWXh"
    >
      <div
        class="MuiPaper-root-96 MuiPaper-elevation1-99 MuiPaper-rounded-97 MuiCard-root-95 sc-gzVnrw gdbKgF"
      >
        <div
          class="sc-dnqmqq fUazxu"
        />
        <form
          class="sc-htoDjs kcChki"
        >
          <div
            aria-label="user-name-field"
            class="MuiInputBase-root-136 MuiInput-root-123 MuiInput-underline-127 MuiInputBase-fullWidth-145 MuiInput-fullWidth-130 sc-iwsKbI fdnPLx"
            paddingtop="2rem"
            required=""
          >
            <input
              aria-label=""
              class="MuiInputBase-input-146 MuiInput-input-131 MuiInputBase-inputType-149 MuiInput-inputType-134"
              placeholder="user name"
              required=""
              type="userName"
              value=""
            />
          </div>
          <div
            aria-label="wages-field"
            class="MuiInputBase-root-136 MuiInput-root-123 MuiInput-underline-127 MuiInputBase-fullWidth-145 MuiInput-fullWidth-130 sc-iwsKbI fdnPLx"
            paddingtop="2rem"
            required=""
          >
            <input
              aria-label=""
              class="MuiInputBase-input-146 MuiInput-input-131 MuiInputBase-inputType-149 MuiInput-inputType-134"
              placeholder="current wages"
              required=""
              type="number"
              value=""
            />
          </div>
          <div
            aria-label="email-field"
            class="MuiInputBase-root-136 MuiInput-root-123 MuiInput-underline-127 MuiInputBase-fullWidth-145 MuiInput-fullWidth-130 sc-iwsKbI fdnPLx"
            paddingtop="2rem"
            required=""
          >
            <input
              aria-label=""
              class="MuiInputBase-input-146 MuiInput-input-131 MuiInputBase-inputType-149 MuiInput-inputType-134"
              placeholder="email"
              required=""
              type="email"
              value=""
            />
          </div>
          <div
            aria-label="password-field"
            class="MuiInputBase-root-136 MuiInput-root-123 MuiInput-underline-127 MuiInputBase-fullWidth-145 MuiInput-fullWidth-130 sc-iwsKbI fdnPLx"
            paddingtop="2rem"
            required=""
          >
            <input
              aria-label=""
              class="MuiInputBase-input-146 MuiInput-input-131 MuiInputBase-inputType-149 MuiInput-inputType-134"
              placeholder="password"
              required=""
              type="password"
              value=""
            />
          </div>
          <div
            aria-label="confirm-password-field"
            class="MuiInputBase-root-136 MuiInput-root-123 MuiInput-underline-127 MuiInputBase-fullWidth-145 MuiInput-fullWidth-130 sc-iwsKbI fdnPLx"
            paddingtop="2rem"
            required=""
          >
            <input
              aria-label=""
              class="MuiInputBase-input-146 MuiInput-input-131 MuiInputBase-inputType-149 MuiInput-inputType-134"
              placeholder="confirm password"
              required=""
              type="password"
              value=""
            />
          </div>
          <div
            class="sc-bdVaJa jsQcrn"
          >
            <button
              aria-label="register-submit-button"
              class="MuiButtonBase-root-179 MuiButton-root-153 MuiButton-contained-164 MuiButton-containedPrimary-165 MuiButton-raised-167 MuiButton-raisedPrimary-168 MuiButton-fullWidth-178 sc-bwzfXH eRNSoX"
              style="margin-top: 2rem;"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label-154"
              >
                Register
              </span>
              <span
                class="MuiTouchRipple-root-182"
              />
            </button>
          </div>
          <footer
            class="sc-gqjmRU vqrqS"
          >
            <span>
              By registering you automatically accept the following:
            </span>
            <a
              href="/register"
            >
              <span>
                Terms and Conditions
              </span>
            </a>
            <a
              href="/register"
            >
              <span>
                Privacy policy
              </span>
            </a>
          </footer>
          <footer
            class="sc-gqjmRU fAqClA"
          >
            Already have an account?
            <a
              href="/"
            >
              Login
            </a>
          </footer>
        </form>
      </div>
    </main>
  </div>
</div>
`;
